### ISSUE TITLE
Gradle to Pom.xml for info.cqframework.cql is wrong

### ISSUE THREAD

**Issue by vitorpamplona** [cqframework/clinical_quality_language#799]
The compile-time dependency on `org.antlr:antlr4` should not be [listed as a runtime dependency](https://mvnrepository.com/artifact/info.cqframework/cql/2.1.0).

The pom file generated from Gradle correctly contains: 
```
  <dependencies>
    <dependency>
      <groupId>org.antlr</groupId>
      <artifactId>antlr4-runtime</artifactId>
      <version>4.10.1</version>
      <scope>compile</scope>
    </dependency>
  </dependencies>
  ```
  
But when [published](https://mvnrepository.com/artifact/info.cqframework/cql/2.1.0), the full-fat `org.antlr:antlr4` dependency came back  

Maybe there was some weird dependency clean up when publishing this version?

**Comment by vitorpamplona** [comment]
Documenting this here as well: 

There's something really strange with the <dependencies> section

[info/cqframework/cql 2.2-SNAPSHOT](https://oss.sonatype.org/service/local/repositories/snapshots/content/info/cqframework/cql/2.2.0-SNAPSHOT/cql-2.2.0-20220927.215651-3.pom) had only one dep: antlr4-runtime
```xml
<dependencies>
  <dependency>
    <groupId>org.antlr</groupId>
    <artifactId>antlr4-runtime</artifactId>
    <version>4.10.1</version>
    <scope>compile</scope>
  </dependency>
</dependencies>
```

But when the final version [info/cqframework/cql 2.2](https://oss.sonatype.org/service/local/repositories/releases/content/info/cqframework/cql/2.2.0/cql-2.2.0.pom) arrives, it gets another dependency on full antlr4
```xml
<dependencies>
  <dependency>
    <groupId>org.antlr</groupId>
    <artifactId>antlr4</artifactId>
    <version>4.10.1</version>
    <scope>compile</scope>
  </dependency>
  <dependency>
    <groupId>org.antlr</groupId>
    <artifactId>antlr4-runtime</artifactId>
    <version>4.10.1</version>
    <scope>compile</scope>
  </dependency>
</dependencies>
```

There was no change in code to justify this.

**Comment by vitorpamplona** [comment]
Apparently, this has been fixed in 2.4.
